src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoDatabaseCustomNode.cs

```diff
@@ -32,18 +32,21 @@ protected override void InitializeChild(TreeNode parent, TreeNode child, object
 
         public override string GetNodeSubType(object smoObject, SmoQueryContext smoContext)
```

Comment:
> public ove [](http://example.com/codeflow?start=7&length=11)

you could still do this for onprem

Reply:
Ledger is available on Azure too so I don't think that's going to be a good long term solution here unfortunately.

Reply:
It would at least allow the functionality to continue as intended for on-prem. The primary impact is on the Ledger database icon. As another data point, in SSMS we don't have the icon enabled for Azure, but we do for on-prem.

Reply:
That's good to know - then going that approach for now here seems reasonable as well to at least keep parity between the two.

Reply:
I can enable it, but I think it still is a performance issue.

Customers who have 200s of databases (from feedback), often see performance hit due to OE connecting to every single database just for the icon. Time it takes is relative to where the database is hosted, and if network is latent, we're going to continue to see timeouts in OE. So is icon worth over OE timing out?

Reply:
SMO doesn't connect to every DB for on-prem servers - it can get that at the server level for all DBs through the sys.databases DMV.

Reply:
Didn't catch that, thanks :)